internal/cli/environment/create/k3d.go

```diff
@@ -0,0 +1,52 @@
+package create
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"capact.io/capact/internal/cli/printer"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/k3d/v4/pkg/client"
+	"github.com/rancher/k3d/v4/pkg/runtimes"
+	k3dtypes "github.com/rancher/k3d/v4/pkg/types"
+	"golang.org/x/sync/errgroup"
+)
+
+// WaitForK3dReadyNodes waits until nodes are in Ready state based on the role message log.
+func WaitForK3dReadyNodes(ctx context.Context, w io.Writer, clusterName string) (err error) {
+	status := printer.NewStatus(w, "")
+	defer func() {
+		status.End(err == nil)
+	}()
+
+	status.Step("Waiting for '%s' cluster nodes to be ready", clusterName)
+
+	runtime := runtimes.SelectedRuntime
+	nodes, err := runtime.GetNodesByLabel(ctx, map[string]string{k3dtypes.LabelClusterName: clusterName})
+	if err != nil {
+		return fmt.Errorf("failed to get nodes for cluster %q", clusterName)
+	}
+
+	nodeWaitGroup, _ := errgroup.WithContext(ctx)
+	for _, node := range nodes {
+		currentNode := node
+		nodeWaitGroup.Go(func() error {
+			readyLogMessage := k3dtypes.ReadyLogMessageByRole[currentNode.Role]
+			if currentNode.Role == k3dtypes.ServerRole {
```

Comment:
it's because of different expectation. I  want to wait for node to be fully up and ready e.g. when I you mount images then wait until they are available.